extensions/Worker.Extensions.CosmosDB/src/Config/CosmosDBBindingOptions.cs

```diff
@@ -23,7 +23,7 @@ internal class CosmosDBBindingOptions
 
         internal ConcurrentDictionary<string, CosmosClient> ClientCache { get; } = new ConcurrentDictionary<string, CosmosClient>();
 
-        internal virtual CosmosClient GetClient(string preferredLocations = "")
+        internal virtual CosmosClient GetClient(CosmosSerializer? serializer = null, string preferredLocations = "")
```

Comment:
Instead of needing to pass `CosmosSerializer` here, you can use DI within the options configuration pipeline (multiple ways to do this), and construct the `CosmosSerializer` then.

Reply:
Something like this?

```csharp
            var cosmosSerializer = new WorkerCosmosSerializer(workerOptions.Value.Serializer);
            applicationBuilder.Services.PostConfigure<CosmosDBBindingOptions>(myOptions =>
            {
                myOptions._serializer = cosmosSerializer;
            });
```

Not sure exactly how to get a hold of WorkerOptions to get the configured serializer

Reply:
I don't like the look of this ☠️

```csharp
            CosmosSerializer? cosmosSerializer = null;
            applicationBuilder.Services.Configure<WorkerOptions>(workerOptions =>
            {
                cosmosSerializer = new WorkerCosmosSerializer(workerOptions.Serializer);
            });
            applicationBuilder.Services.PostConfigure<CosmosDBBindingOptions>(myOptions =>
            {
                myOptions._serializer = cosmosSerializer;
            });
```

Reply:
Option 1: via delegates

``` CSharp
applicationBuilder.Services.AddOptions<CosmosDBBindingOptions>()
    .Configure<IOptions<WorkerOptions>>((cosmosOptions, workerOptions) => /* access to both here */);
```

Option 2: via type

``` CSharp
internal class ConfigureCosmosDBBindingOptions : IConfigureOptions<CosmosDBBindingOptions>
{
    public ConfigureCosmosDBBindingOptions(IOptions<WorkerOptions> workerOptions)
    {
       // ....
    }

    public void Configure(CosmosDBBindingOptions options)
    {
        // set serializer here.
    }
}


applicationBuilder.Services.ConfigureOptions<ConfigureCosmosDBBindingOptions>();
```

Reply:
Thanks for this explanation, this helps a lot